src/state_machine.zig

```diff
@@ -594,6 +745,653 @@ test "linked accounts" {
+    try testing.expectEqual(@as(u64, 0), state_machine.get_account(7).?.*.debits_reserved);
+    try testing.expectEqual(@as(u64, 0), state_machine.get_account(8).?.*.credits_reserved);
+    try testing.expect(state_machine.get_transfer(tran_test_vector[22].input.id) == null);
+}
+
+test "create/lookup/rollback commits" {
+    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+    defer arena.deinit();
+    const allocator = &arena.allocator;
+
+    const CommitTestVector = struct { outcome: CommitTransferResult, input: Commit };
+
+    var accounts = [_]Account{
+        std.mem.zeroInit(Account, .{ .id = 1 }),
+        std.mem.zeroInit(Account, .{ .id = 2 }),
+        std.mem.zeroInit(Account, .{ .id = 3 }),
+        std.mem.zeroInit(Account, .{ .id = 4 }),
+    };
+
+    var transfers = [_]Transfer{
+        std.mem.zeroInit(Transfer, .{
+            .id = 1,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 2,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+            .flags = .{ .two_phase_commit = true },
+            .timeout = 25,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 3,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+            .flags = .{ .two_phase_commit = true },
+            .timeout = 25,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 4,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+            .flags = .{ .two_phase_commit = true },
+            .timeout = 1,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 5,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+            .flags = .{
+                .two_phase_commit = true,
+                .condition = true,
+            },
+            .timeout = 25,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 6,
+            .amount = 15,
+            .debit_account_id = 1,
+            .credit_account_id = 2,
+            .flags = .{
+                .two_phase_commit = true,
+                .condition = false,
+            },
+            .timeout = 25,
+        }),
+        std.mem.zeroInit(Transfer, .{
+            .id = 7,
+            .amount = 15,
+            .debit_account_id = 3,
+            .credit_account_id = 4,
+            .flags = .{ .two_phase_commit = true },
+            .timeout = 25,
+        }),
+    };
+
+    var state_machine = try StateMachine.init(allocator, accounts.len, transfers.len, 1);
+    defer state_machine.deinit();
+
+    const input = std.mem.asBytes(&accounts);
+    const output = try allocator.alloc(u8, 4096);
+
+    // Use a timestamp of 0 since this is just a test
```

Comment:
We can drop this comment, and just inline the timestamp of 0 when calling `prepare()`